mlir/include/mlir/Dialect/EmitC/IR/EmitC.td

```diff
@@ -51,7 +51,8 @@ class EmitC_BinaryOp<string mnemonic, list<Trait> traits = []> :
 def CExpression : NativeOpTrait<"emitc::CExpression">;
 
 // Types only used in binary arithmetic operations.
-def IntegerIndexOrOpaqueType : AnyTypeOf<[EmitCIntegerType, Index, EmitC_OpaqueType]>;
+def IntegerIndexOrOpaqueType : AnyTypeOf<[EmitCIntegerType, Index,
+  EmitC_SignedSizeT, EmitC_SizeT, EmitC_OpaqueType]>;
```

Comment:
We have type predicates for Integer types [here](https://github.com/llvm/llvm-project/blob/84729c9df30b29d5f4e903ad71235a6aa0c764d6/mlir/include/mlir/Dialect/EmitC/IR/EmitCTypes.td#L25). I would add an EmitCSizeTType predicate there as well which calls `isAnySizeTType` and use it here.

Can you also add a few tests for operations that use this predicate, like `emitc.add` for example.

(I'm wondering why we have this duplicated with `emitc::isIntegerIndexOrOpaqueType` though. Did we forget to change this to a predicate here? @TinaAMD, @mgehre-amd

Reply:
I went ahead and replaced this with the predicate. It looks like it was added in [https://github.com/llvm/llvm-project/commit/1f268092c7af20c21d4a594678b647cab050602a] and used in the `verify()` functions, where the `.td` type specifications are not available. I didn't find updates using `emitc::isIntegerIndexOrOpaqueType` and mentioning not updating the `.td` at the same time.  If there was a more fundamental reason the predicate wasn't used, I'll revert this.